lib/propolis/src/instance.rs

```diff
@@ -397,6 +414,7 @@ impl Instance {
         &self,
         migrate_ctx_id: CtxId,
         pause_chan: std::sync::mpsc::Receiver<()>,
+        on_pause_reqs_sent: Option<Box<PauseRequestsSentFunc>>,
     ) -> Result<(), TransitionError> {
```

Comment:
Sweet, thanks for looking into this! The only suggestion I have is just making `migrate_pause` directly return `tokio::sync::Notify` the caller can choose to wait on. The more generalized, callback based interface here is nice but the only callers we have don't really need it. Did you have another use case in mind?

Reply:
Funny you should ask :) I did exactly that in an earlier draft on Friday, but woke up this morning and felt a little squicky about having a non-`async` (and nominally environment-agnostic) routine start returning `tokio` primitives, so I switched it to something that lets the caller pick the notification discipline.

On the other hand, you're right that this code is only called in a couple of specialized places, and they already uphold the "I could easily `await` this" invariant. And it is simpler just to return the `Notify`. (And I'm about to revisit a bunch of this anyway to help deal with #155.) 

I'll change this over presently (I also want to rearrange things a bit so that the `notify` call/callback information lives with the other implicit state transition actions and not in the routine that otherwise just talks to entities).

Reply:
Should be fixed (and the notification site tidied up) in 6ab17d5.